tutor/src/screens/assignment-edit/homework/choose-exercises.jsx

```diff
@@ -51,8 +53,14 @@ class ChooseExercises extends React.Component {
     this.onDisplayAddEditQuestionModal(ev, true);
   }
 
+  @action.bound onDeleteExercise() {
+    this.props.ux.exercises.deleteExercise(this.props.ux.course, this.selectedExercise);
+    this.showDeleteQuestionModal = false;
+  }
+
   getExerciseActions = (exercise) => {
     const { ux } = this.props;
+    const isUserGeneratedQuestion = exercise.belongsToCurrentUserProfileId(User.profile_id);
```

Comment:
@nathanstitt Once you have your other PR merged, I can change it to the new cleaner code you have.